core/src/main/java/io/grpc/internal/InternalSubchannel.java

```diff
@@ -573,7 +575,7 @@ public void run() {
           }
         }
       });
-      Preconditions.checkState(activeTransport != transport,
+      Preconditions.checkState(shutdownInitiated || activeTransport != transport,
```

Comment:
Just check `shutdownInitiated`, and you can do it at the beginning of the method.

Reply:
thanks done. had to fix few tests PTAL